### ISSUE TITLE
Compatibility with Elasticsearch 5.0.0

### ISSUE THREAD

**Issue by Sennik** [elasticquent/Elasticquent#124]
`illegal_argument_exception: The parameter [fields] is no longer supported, please use [stored_fields] to retrieve stored fields or _source filtering if the field is not stored`

It seems that Elastiquent is not compatible with the latest version of Elasticsearch. Or did I something wrong?

**Comment by niefufeng** [comment]
`elasticsearch` version 5.x 不再支持`fields`字段

**Comment by niefufeng** [comment]
@timgws @chuangbo

**Comment by timgws** [comment]
I will be installing ES 5 soon and having a bit of a play around with it to work out exactly why this is not working.

**Comment by ellisio** [comment]
Any progress on this?

**Comment by andrewmclagan** [comment]
HAs there been any progress with ES 5.x.x ?

**Comment by timgws** [comment]
There has been a tiny bit of progress, @andrewmclagan

**Comment by andrewmclagan** [comment]
Awesome, Seems the underlying library `elasticsearch/elasticsearch` needs updating to `5.x.x`. I would love to give a hand, is there a branch that I can take a look at. I will be needing 5 support ASAP and I'm willing to jump straight into it tomorrow.

@timgws

**Comment by timgws** [comment]
Awesome. Let me commit some code later tonight so you can test tomorrow and start digging into it straight away in the morning, @andrewmclagan

**Comment by andrewmclagan** [comment]
@timgws appreciate it :-) just let me know what the branch is.

**Comment by jacob-v-dam** [comment]
When will this be released? I need this for a new version of elasticsearch.

**Comment by ppeer** [comment]
I am running elasticsearch 5.2.2 & in combination with elasticquent and the searchQuery command it returns:
 `illegal_argument_exception: The parameter [fields] is no longer supported, please use [stored_fields] to retrieve stored fields or _source filtering if the field is not stored`

is there a solution yet or do you recommend installing an older elasticsearch version?

**Comment by andrewmclagan** [comment]
I was going to submit a PR to address this although we have migrated to our own solution. Personally I think this repo is great, although it needs a little more abstraction and separation for our needs. Apologies everyone.

Maybe at a later date we can migrate some of our work into this repo.

**Comment by ppeer** [comment]
Lacking the time to give input to this great repo, i switched back to ES 2.4 to overcome this problem. For now I hope elasticquent will try to keep up it's intermediate position between (new) laravel and elasticsearch (versions).

**Comment by zhangyuxiao** [comment]
Any progress on this?

**Comment by timgws** [comment]
Yes, there is. I will be pushing the new version for testing soon.

I am currently working on a test suite that can be used to test the application end-to-end. This will help ease issues with updating to newer versions of Elasticsearch & compatibility with this application 👍

**Comment by andrewmclagan** [comment]
@timgws I would be interested in how you are undertaking the e2e tests!

We use a docker based build pipeline called Drone, very easy to boot up an Elasticsearch server and then tear it down again.

**Comment by timgws** [comment]
@andrewmclagan Currently, it's based on https://github.com/elastic/elasticsearch-docker, a modified `Makefile` with a modified [elasticsearch.yaml](https://github.com/elastic/elasticsearch-docker/blob/master/build/elasticsearch/elasticsearch.yml) file depending on the passed in values for version & port in the Makefile.

I am testing against Elasticsearch 5.4, 2.4 and 2.0.

My `test.sh` script checks if an Elasticsearch image has been built for the selected version, builds it if not, then runs all the images on different ports.

PHPUnit then runs the same test suite three times (one time for each port that Elasticsearch is running on). It reports the latest version of Elasticsearch to coveralls.

I am not yet sure how I can set up Coveralls to run the test suite automatically for all versions of Elasticsearch. This would be preferable.

Have just performed a quick Google for Drone and so far this looks like it might be the missing piece of the puzzle!

I am currently busy working on another project, so for the time being the test.sh script that I have is going to have to do. I will try and get that all checked in soon!

**Comment by andrewmclagan** [comment]
Drone is amazing...

**Comment by musab95** [comment]
I am running elasticsearch 5.4.0 with elasticquent in which search() is work but searchByQuery() return error:
exception 'Elasticsearch\Common\Exceptions\BadRequest400Exception' with message '{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"The parameter [fields] is no longer supported, please use [stored_fields] to retrieve stored fields or _source filtering if the field is not stored"}],"type":"illegal_argument_exception","reason":"The parameter [fields] is no longer supported, please use [stored_fields] to retrieve stored fields or _source filtering if the field is not stored"},"status":400}'

Please suggest me some solution.

**Comment by rschuil** [comment]
@musab95 not a solution but as a temporary workaround I added this to my model:
`  private function buildFieldsParameter($getSourceIfPossible, $getTimestampIfPossible)
    {
        return null;
    }
`

**Comment by musab95** [comment]
thanks @rschuil 
its work for me

**Comment by ppeer** [comment]
unfortunately, not working for me: laravel 5.2 elasticsearch 5.3.0. Is there anything else I could try?

**Comment by ajayfroiden** [comment]
Can anyone please let me know if it now available for elasticsearch 5.4.0 or not? I could not find the clear answer.

**Comment by Its-rahulG** [comment]
hi, is there any update regarding this thread, i'm dying to include this package in my production app.

**Comment by molbi** [comment]
@timgws hi, we use this package a lot, it is great, but know, we need upgrade ES to v5 and i really dont want using fork repositories, but know it is only one option. Do you know, when new patch will be? ;) thank you man.

**Comment by ajayfroiden** [comment]
@timgws and other contributors. Please respond the the queries. We do want this package to be upgraded

**Comment by PeteJStewart** [comment]
@timgws Did this get resolved?
From what I can tell, Elasticquent doesn't work with Elasticsearch 5.x?
It only seems to work with 2.x.
Is this the case?  Is there a plan to resolve this, or should we all be looking at alternative systems?

**Comment by xav-akumar8** [comment]
@timgws Please provide latest in order to resolve this

**Comment by jeversen** [comment]
@molbi Did you find a workaround or did you fork this repo and fixed the ES5 issue ?

@timgws any news on this ?

Would really like to use this package in production too, but going further with ES 2.x doesn't seem to be acceptable.

**Comment by jeversen** [comment]
Moved my codebase to [https://github.com/sleimanx2/plastic]. Seems to fit my needs and can be used by the newer ES versions.

**Comment by nospoon** [comment]
Given there doesn't seem to be any interest in fixing the issue, it would seem to me the project is dead?

**Comment by ssayen** [comment]
Thanks @jeversen switched to Plastic too. Works just fine and migration from ElasticQuent is easy. Too bad, this project was really nice.

**Comment by coffeeburrito** [comment]
In case anyone is curious, the changes to support ES 5.x are actually pretty minor and there's a few forks with [working code](https://github.com/zoul0813/Elasticquent). Combined with the [ability to override composer dependencies with a vcs repo](http://mnapoli.fr/overriding-dependencies-with-composer), that was all I needed to get to ES5. It'll do for now, but hopefully Elasticquent gets an official update before too long.

In my own fork I bumped the elasticsearch version to 6.x and found the same code worked without any further changes, although there's probably new features it's not taking advantage of.
